akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdPublisher.scala

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.query.journal.leveldb
+
+import scala.concurrent.duration._
+import akka.actor.ActorLogging
+import akka.actor.ActorRef
+import akka.actor.Props
+import akka.persistence.JournalProtocol._
+import akka.persistence.Persistence
+import akka.stream.actor.ActorPublisher
+import akka.stream.actor.ActorPublisherMessage.Cancel
+import akka.stream.actor.ActorPublisherMessage.Request
+import akka.persistence.journal.leveldb.LeveldbJournal
+import akka.persistence.query.EventEnvelope
+
+/**
+ * INTERNAL API
+ */
+private[akka] object EventsByPersistenceIdPublisher {
+  def props(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long, refreshInterval: Option[FiniteDuration],
+            maxBufSize: Int, writeJournalPluginId: String): Props =
+    Props(new EventsByPersistenceIdPublisher(persistenceId, fromSequenceNr, toSequenceNr, refreshInterval,
+      maxBufSize, writeJournalPluginId))
+
+  private case object Continue
+}
+
+class EventsByPersistenceIdPublisher(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long,
+                                     refreshInterval: Option[FiniteDuration],
+                                     maxBufSize: Int, writeJournalPluginId: String)
+  extends ActorPublisher[EventEnvelope] with DeliveryBuffer[EventEnvelope] with ActorLogging {
+  import EventsByPersistenceIdPublisher._
+
+  val journal: ActorRef = Persistence(context.system).journalFor(writeJournalPluginId)
+
+  var currSeqNo = fromSequenceNr
+
+  val tickTask = refreshInterval.map { interval ⇒
+    import context.dispatcher
+    context.system.scheduler.schedule(interval, interval, self, Continue)
+  }
+
+  override def postStop(): Unit = {
+    tickTask.foreach(_.cancel())
+  }
+
+  def receive = init
+
+  def init: Receive = {
+    case _: Request ⇒
+      journal ! LeveldbJournal.SubscribePersistenceId(persistenceId)
+      replay()
+    case Continue ⇒ // skip, wait for first Request
+    case Cancel   ⇒ context.stop(self)
+  }
+
+  def idle: Receive = {
+    case Continue | _: LeveldbJournal.Changed ⇒
+      if (timeForReplay)
+        replay()
+
+    case _: Request ⇒
+      deliverBuf()
+      if (refreshInterval.isEmpty) {
```

Comment:
that's not the same thing, then it must be 

    if (refreshInterval.isEmpty && buf.isEmpty) onCompleteThenStop()
    else if (refreshInterval.isEmpty) self ! Continue